qiskit/circuit/register.py

```diff
@@ -17,217 +17,7 @@
-            CircuitError: if the `key` is not an integer or not in the range `(0, self.size)`.
-        """
-        if not isinstance(key, (int, np.integer, slice, list)):
-            raise CircuitError("expected integer or slice index into register")
-        if isinstance(key, slice):
-            return self._bits[key]
-        elif isinstance(key, list):  # list of qubit indices
-            if max(key) < len(self):
-                return [self._bits[idx] for idx in key]
-            else:
-                raise CircuitError("register index out of range")
-        else:
-            return self._bits[key]
-
-    def __iter__(self):
-        for idx in range(self._size):
-            yield self._bits[idx]
-
-    def __contains__(self, bit):
-        if self._bit_indices is None:
-            self._bit_indices = {bit: idx for idx, bit in enumerate(self._bits)}
-
-        return bit in self._bit_indices
-
-    def index(self, bit):
-        """Find the index of the provided bit within this register."""
-        if self._bit_indices is None:
-            self._bit_indices = {bit: idx for idx, bit in enumerate(self._bits)}
-
-        try:
-            return self._bit_indices[bit]
-        except KeyError as err:
-            raise ValueError(f"Bit {bit} not found in Register {self}.") from err
-
-    def __eq__(self, other):
-        """Two Registers are the same if they are of the same type
-        (i.e. quantum/classical), and have the same name and size. Additionally,
-        if either Register contains new-style bits, the bits in both registers
-        will be checked for pairwise equality. If two registers are equal,
-        they will have behave identically when specified as circuit args.
-
-        Args:
-            other (Register): other Register
-
-        Returns:
-            bool: `self` and `other` are equal.
-        """
-        if self is other:
-            return True
-
-        res = False
-        if (
-            type(self) is type(other)
-            and self._repr == other._repr
-            and all(
-                # For new-style bits, check bitwise equality.
-                sbit == obit
-                for sbit, obit in zip(self, other)
-                if None in (sbit._register, sbit._index, obit._register, obit._index)
-            )
-        ):
-            res = True
-        return res
-
-    def __hash__(self):
-        """Make object hashable, based on the name and size to hash."""
-        return self._hash
-
-    def __getstate__(self):
-        # Specifically exclude _bit_indices from pickled state as bit hashes
-        # can in general depend on the hash of their containing register,
-        # which may not have yet been initialized.
-        return self._name, self._size, self._hash, self._repr, self._bits
-
-    def __setstate__(self, state):
-        self._name, self._size, self._hash, self._repr, self._bits = state
-        self._bit_indices = None
+Register = circuit.Register
```

Comment:
Why not just do: `from qiskit._accelerate.circuit import Register` ?

Reply:
I was following what we do for importing `CircuitInstruction`, no reason other than that

Reply:
I got rid of the stub files in 508751975.